gammacat/dataset.py

```diff
@@ -0,0 +1,47 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import logging
+from .utils import load_yaml, write_yaml
+from gammapy.catalog.gammacat import GammaCatResource
+
+__all__ = [
+    'DataSet',
+]
+
+log = logging.getLogger(__name__)
+
+class DataSet:
+
+    resource_type = 'ds'
+
+    def __init__(self, filename):
+        log.debug('DataSet.__init__()')
+        self.filename = filename
+        self.data = load_yaml(filename)
+
+    def write(self, filename):
+        write_yaml(self.data, filename)
+
+    def reference_id(self):
+        return self.data['reference_id']
+
+    def folder(self):
+        return self.data['reference_id'].replace('&', '%26')
+
+    def source_id(self):
+        return self.data['source_id']
+
+    def to_dict(self):
+        return self.data
+
+    def to_GammaCatResource(self):
+        try:
```

Comment:
this try except can be written equivalently in one line:
```
file_id = self.data.get('file_id', -1)
```
Why is a special case needed, what's -1?

Reply:
Because in the light curve and see files file_id is set to -1 if there is only one file.